pkg/serverless/metrics/enhanced_metrics_test.go

```diff
@@ -9,154 +9,160 @@ import (
 	}, {
 		Name:       initDurationMetric,
 		Value:      0.1,
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}})
 }
 
 func TestGenerateEnhancedMetricsFromReportLogNoColdStart(t *testing.T) {
-	metricsChan := make(chan []metrics.MetricSample)
+	demux := aggregator.InitTestAgentDemultiplexerWithFlushInterval(time.Hour)
+	defer demux.Stop(false)
 	tags := []string{"functionname:test-function"}
 	reportLogTime := time.Now()
 
-	go GenerateEnhancedMetricsFromReportLog(0, 1000.0, 800.0, 1024.0, 256.0, reportLogTime, tags, metricsChan)
+	go GenerateEnhancedMetricsFromReportLog(0, 1000.0, 800.0, 1024.0, 256.0, reportLogTime, tags, demux)
 
-	generatedMetrics := <-metricsChan
+	generatedMetrics := demux.WaitForSamples(100 * time.Millisecond)
 
-	assert.Equal(t, generatedMetrics, []metrics.MetricSample{{
+	assert.Equal(t, generatedMetrics[:5], []metrics.MetricSample{{
 		Name:       maxMemoryUsedMetric,
 		Value:      256.0,
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}, {
 		Name:       memorySizeMetric,
 		Value:      1024.0,
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}, {
 		Name:       billedDurationMetric,
 		Value:      0.80,
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}, {
 		Name:       durationMetric,
 		Value:      1.0,
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}, {
 		Name:       estimatedCostMetric,
 		Value:      calculateEstimatedCost(800.0, 1024.0),
 		Mtype:      metrics.DistributionType,
 		Tags:       tags,
 		SampleRate: 1,
-		Timestamp:  float64(reportLogTime.UnixNano()),
+		Timestamp:  float64(reportLogTime.UnixNano()) / float64(time.Second),
 	}})
 }
 
 func TestSendTimeoutEnhancedMetric(t *testing.T) {
-	metricsChan := make(chan []metrics.MetricSample)
+	demux := aggregator.InitTestAgentDemultiplexerWithFlushInterval(time.Hour)
+	defer demux.Stop(false)
 	tags := []string{"functionname:test-function"}
 
-	go SendTimeoutEnhancedMetric(tags, metricsChan)
+	go SendTimeoutEnhancedMetric(tags, demux)
 
-	generatedMetrics := <-metricsChan
+	generatedMetrics := demux.WaitForSamples(100 * time.Millisecond)
 
-	assert.Equal(t, generatedMetrics, []metrics.MetricSample{{
+	assert.Equal(t, generatedMetrics[:1], []metrics.MetricSample{{
```

Comment:
It would be nice to check if `generatedMetrics` is not empty

Reply:
I think this should not have been committed, I'll look into this.